lib/Implicit/impSurface.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2001-2010  Terence M. Welsh
+ *
+ * This file is part of Implicit.
+ *
+ * Implicit is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * Implicit is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef IMPSURFACE_H
+#define IMPSURFACE_H
+
+#include <functional>
+
+#ifdef WIN32
+#include <windows.h>
+#endif
+#include <vector>
+
+#include <kodi/gui/gl/GL.h>
```

Comment:
Why does implicit lib include Kodi headers?

Reply:
You mean since they are deposited in the addon?

If so, they are temporarily stored in addon and will be added to Kodi M * and removed here.

Reply:
No my question is not why `kodi/gui/gl/GL.h` and all the other files in lib/kodi are checked into this repo.
My question is, why does a library that doesn't know anything about Kodi, include a Kodi header file?

Reply:
On the one hand, a lot of changes were needed to make GL4 compatible. The other is, this was designed purely for GL under Windows and therefore had no GLES support. By using the Kodi header, the required one is inserted.

This library can be seen as a death part of the original, so I'm not ashamed to make this suitable for us.
https://github.com/reallyslickscreensavers/rslibs/tree/master/Implicit